app/controllers/writable_controller.rb

```diff
@@ -66,8 +67,11 @@ def show_post(cur_page=nil)
       self.page = cur_page = cur_page.to_i
     end
 
+    read_at = @post.read_time_for(@replies) if logged_in?
```

Comment:
Why did this get moved? I was reasonably sure this needed to happen after paginate?

Reply:
...looking at the intervening code it might be so it correctly includes the written in the replies to check for read status. otoh you probably... want pagination there, yes. hm.